src/Mvc/Mvc.Core/src/Infrastructure/ActionSelectionTable.cs

```diff
@@ -66,10 +66,11 @@ public static ActionSelectionTable<ActionDescriptor> Create(ActionDescriptorColl
                 // For action selection, ignore attribute routed actions
                 items: actions.Items.Where(a => a.AttributeRouteInfo == null),
 
-                getRouteKeys: a => a.RouteValues.Keys,
+                getRouteKeys: a => a.RouteValues?.Keys,
```

Comment:
Encountered when adding unit tests. These needed to be hardened against potentially null values.